### PR TITLE
[onert] Change the batch_size of input tensor in train_prepare

### PR DESCRIPTION
**Pull request by jyoungyun**
This commit changes the batch_size of input tensor in train_prepare
nnfw internal api. Other operation will be changed the tensor info
through Static shape inference function.

ONE-DCO-1.0-Signed-off-by: Jiyoung Yun <jy910.yun@samsung.com>

### CODE REVIEW

File: runtime/onert/api/src/nnfw_api_internal.cc [hunk 1]
```diff
@@ -1156,6 +1156,18 @@ NNFW_STATUS nnfw_session::train_prepare(const nnfw_train_info *info)
 
   try
   {
+    for (uint32_t i = 0; i < getInputSize(); ++i)
+    {
+      auto info = _nnpkg->inputInfo(i);
+      onert::ir::Shape new_shape(info.shape());
+      for (int32_t i = 0; i < new_shape.rank(); i++)
```
Comment (ragmani) [inline runtime/onert/api/src/nnfw_api_internal.cc:1163]: I think this for loop is unnecessary. And the local variable `i` is duplicated.

Comment (jyoungyun) [reply runtime/onert/api/src/nnfw_api_internal.cc]: Ah sorry it's my mistake. :(

**Review by ragmani — DISMISSED** [review]
LGTM

File: runtime/onert/api/src/nnfw_api_internal.cc [hunk 2]
```diff
@@ -1156,6 +1156,15 @@ NNFW_STATUS nnfw_session::train_prepare(const nnfw_train_info *info)
 
   try
   {
+    for (uint32_t i = 0; i < getInputSize(); ++i)
+    {
+      auto info = _nnpkg->inputInfo(i);
+      onert::ir::Shape new_shape(info.shape());
+      // TODO Consider batch size index
+      new_shape.dim(0) = training_info.batchSize();
```
[the remark this answers is not in the document]
Comment (jyoungyun) [reply runtime/onert/api/src/nnfw_api_internal.cc:1164]: Yes. For now, we assume that the batch size is always the first dim.

[the remark this answers is not in the document]
Comment (glistening) [reply runtime/onert/api/src/nnfw_api_internal.cc:1164]: @jyoungyun As I understand your code above, it is the opposite. It assumes the first dim is always overwritten with `batch_size`. I am not sure, it is safe when dim[0] != 1.

[the remark this answers is not in the document]
Comment (jyoungyun) [reply runtime/onert/api/src/nnfw_api_internal.cc:1166]: I added context to compare dim(0) != 1. And I also updated the commit message.

**Review by glistening — APPROVED** [review]
LGTM

**Review by ragmani — APPROVED** [review]

**Pull request merged**